packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -87,7 +87,32 @@ export default function astro({ config, devServer }: AstroPluginOptions): vite.P
           map,
         };
       } catch (err: any) {
-        // if esbuild threw the error, find original code source to display (if it’s mapped)
+        // improve compiler errors
+        if (err.toString() === 'RuntimeError: unreachable') {
+          const search = new URLSearchParams({
+            labels: 'compiler',
+            title: '🐛 BUG: `@astrojs/compiler` panic',
+            body: `### Describe the Bug
+
+\`@astrojs/compiler\` encountered an unrecoverable error when compiling the following file.
+
+**${id.replace(fileURLToPath(config.projectRoot), '')}**
+\`\`\`astro
+${source}
```

Comment:
This is where we save the actual code and pass along in a URL (all @natemoo-re’s idea!)